pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -166,6 +178,26 @@ contract WeightedPool is IWeightedPool, BalancerPoolToken, PoolInfo, Version {
         } else {
             return normalizedWeights;
         }
+        if (numTokens > 4) {
+            normalizedWeights[4] = _normalizedWeight4;
+        } else {
+            return normalizedWeights;
+        }
+        if (numTokens > 5) {
+            normalizedWeights[5] = _normalizedWeight5;
+        } else {
+            return normalizedWeights;
+        }
+        if (numTokens > 6) {
+            normalizedWeights[6] = _normalizedWeight6;
+        } else {
+            return normalizedWeights;
+        }
+        if (numTokens > 7) {
+            normalizedWeights[7] = _normalizedWeight7;
+        } else {
+            return normalizedWeights;
+        }
```

Comment:
This one's not needed, but also not wrong.

Reply:
Made it look more like v2, and removed the last one.